src/Bicep.Core/TypeSystem/CyclicCheckVisitor.cs

```diff
@@ -15,19 +13,17 @@ namespace Bicep.Core.TypeSystem
 {
     public sealed class CyclicCheckVisitor : SyntaxVisitor
     {
-        private readonly IReadOnlyDictionary<string, DeclaredSymbol> declarations;
```

Comment:
The problem here is that we can no longer assume each resource has a unique symbol name. This was the one piece of code I came across that has this assumption - but in general it's no longer valid. Turns out this wasn't a big change for this class.